### ISSUE TITLE
[2.7.3] Required field not being filled causes js error, without showing user they need to fix a required field

### ISSUE THREAD

**Issue by Mark-H** [modxcms/revolution#15088]
### Summary

A required field when editing a resource not being filled causes a javascript error, and prevents the "Please correct the errors in your form before submitting" error from being shown to the user.

The specific error is as follows:

```
modx.panel.resource.js:251 Uncaught TypeError: Cannot read property 'hidden' of undefined
    at MODx.panel.Resource.failureSubmit (modx.panel.resource.js:251)
    at h.Event.fire (ext-all.js:21)
    at MODx.panel.Resource.fireEvent (ext-all.js:21)
    at MODx.toolbar.ActionButtons.handleClick (modx.component.js:298)
    at S.onClick (ext-all.js:21)
    at HTMLSpanElement.I (ext-all.js:21)
```

Also discussed [on the forum](https://community.modx.com/t/save-button-for-resources-not-functioning/2654) and modmore ticket #22940. 

### Step to reproduce

I've not reduced the identified behavior to a very narrow use case, but I suspect this happens when you leave any TV that is marked as required (allowBlank = no) empty. In this particular case, it was a checkbox tv that required a value and had an invalid default value (default value not in the options) and therefore always invalid. 

It seems this may also require the TVs section at the bottom of the page instead of in the tab (`tvs_below_content` setting enabled), but I've not verified that yet. See below.

Trying to save will mark the field in red to indicate it's required, however if that's on a different tab or way at the bottom of the page, that's near impossible to spot. The lack of a notice is definitely troubling.

The problem seems to be in the `failureSubmit` method on `MODx.panel.Resource`. 

``` js
        // The next check needs the tabs index value to check if it is hidden or not
        if (tabs && tabs.items && tabs.items.keys) {
            tabs_index = tabs.items.keys.indexOf(tab_name);
        }

        // We are already on a tab that has an invalid state. No need to switch
        if (!tabs.items.items[tabs_index].hidden)  { // <--- breaks here on accessing .hidden on undefined
            return;
        }
```

It appears that `tabs_index` is -1 (i.e. the current tab, `tab_name = "modx-panel-resource-tv"`, does not exist in the tab panel). 

Because it breaks on what would normally be the tvs tab, I strongly suspect this only happens when tvs_below_content is enabled as it then no longer is a tab, but a separate panel at the bottom. 

### Possible fix

A simple workaround would be to make sure `tabs.items.items[tabs_index]` exists before trying to access the `hidden` property. A simple check like this:

```
 if (!tabs.items.items[tabs_index] || tabs.items.items[tabs_index].hidden)  {
```

That said, it's probably worthwhile figuring out if it's needed for this logic to take into account the separate TV panel below the content and perhaps open the appropriate (sub)tab there if it doesn't already do that. 

### Observed behavior

The error mentioned above.

### Expected behavior

No error, and the `"Please correct the errors in your form before submitting."` popup should show indicating a field was marked as invalid. 

### Environment

MODX 2.7.3 on (AFAIK) any browser.

**Comment by rthrash** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/save-button-for-resources-not-functioning/2654/28

**Comment by Ruslan-Aleev** [comment]
If there is a tab with required TV, then there are no errors, the tab is activated:

![tv_req](https://user-images.githubusercontent.com/12523676/85199348-6c306a80-b2f7-11ea-98be-366aefffa85f.gif)

But, if the TV is located in the category (sub-tab), then sub-tab activation does not happening (related https://github.com/modxcms/revolution/issues/11112):

![tv_req_2](https://user-images.githubusercontent.com/12523676/85199444-57080b80-b2f8-11ea-98d2-b6016d66054a.gif)

As I understand it, this bug arose after the implementation of this PR - https://github.com/modxcms/revolution/pull/14235.

By the way, there is another problem with activating tabs, it does not always work - https://github.com/modxcms/revolution/issues/15116.

**Comment by Ruslan-Aleev** [comment]
By the way, it would also be useful if the screen would scroll to the elements with an error - https://github.com/modxcms/revolution/issues/14743.
Ideally, several related problems can be fixed with one PR and significantly improve UX :)

**Comment by Mark-H** [comment]
Fixed in 2.8
